File: src/main/java/endytkn/randomEvents/entities/GenericEntity.java
package endytkn.randomEvents.entities;

import net.minecraft.world.entity.EntityType;
import net.minecraft.world.entity.PathfinderMob;
import net.minecraft.world.level.Level;

public class GenericEntity<T extends PathfinderMob> extends PathfinderMob {
    public GenericEntity(EntityType<T> entityType, Level level) {
        super(entityType, level);
    }
}

File: src/main/kotlin/endytkn/randomEvents/entities/Generic.entity.kt
package endytkn.randomEvents.entities

import net.minecraft.world.entity.EntityType
import net.minecraft.world.entity.PathfinderMob
import net.minecraft.world.level.Level

class GenericEntity<T : PathfinderMob>(entityType: EntityType<T>, level: Level): PathfinderMob(entityType, level) {

}